### ISSUE TITLE
Change Request: Disable low concurrency warning

### ISSUE THREAD

**Issue by null77** [eslint/eslint#20343]
### ESLint version

v9.34

### What problem do you want to solve?

I'm seeing the low concurrency warning when I invoke linting, on an incremental lint that has most of the files cached, using 'auto' for the concurrency setting.

### What do you think is the correct solution?

I'd like a way to disable this warning. I don't mind low concurrency, and there's no way to determine up front how many files are left to lint. I really want to take advantage of concurrency and I don't want my devs to get worried about this warning.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Thanks for the feedback. Is there a reason you don't turn concurrency off? If you're getting the warning, it means it's not benefiting you.

**Comment by null77** [comment]
So it would benefit us in scenarios where many files are dirty.. i.e. there's some kind of configuration change and lint needs to refresh the whole project. But there's no way of knowing when you invoke the command if you're refreshing the whole project or just one file, etc. So the warning hits us in the second case without any way to prevent it. Does that make sense?

**Comment by fasttime** [comment]
In current versions of Node.js, you can disable the warning with the runtime flag [`--disable-warning`](https://nodejs.org/api/cli.html#--disable-warningcode-or-type) and the warning type. For example, when starting ESLint in a Unix shell:

```shell
NODE_OPTIONS="--disable-warning=ESLintPoorConcurrencyWarning" npx eslint --concurrency=auto
```

Also note that the warning indicates concurrency is likely too high, not too low.

**Comment by null77** [comment]
That works.
